src/main/java/com/microsoft/sqlserver/jdbc/dtv.java

```diff
@@ -1083,8 +1083,16 @@ void execute(DTV dtv, Double doubleValue) throws SQLServerException {
 
         void execute(DTV dtv, BigDecimal bigDecimalValue) throws SQLServerException {
             if (DDC.exceedsMaxRPCDecimalPrecisionOrScale(bigDecimalValue)) {
-                String strValue = bigDecimalValue.toString();
-                tdsWriter.writeRPCStringUnicode(name, strValue, isOutParam, collation);
+                if (JDBCType.DECIMAL == dtv.getJdbcType() || JDBCType.NUMERIC == dtv.getJdbcType()) {
+                    // Throw exception for DECIMAL and NUMERIC Datatypes
+                    MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_valueOutOfRangeSQLType"));
+                    Object[] msgArgs = {JDBCType.DECIMAL == dtv.getJdbcType() ? SSType.DECIMAL : SSType.NUMERIC};
```

Comment:
This line just just be:
`Object[] msgArgs = {dtv.getJdbcType};`

Reply:
`dtv.getJdbcType()` will return `"java.math.BigDecimal"` for both `JDBCType.DECIMAL` and `JDBCType.NUMERIC` - which is not what we want to throw in error message. We are trying to specify SQLType used for parameter setter.

Reply:
Does it really return `"java.math.BigDecimal"`? Please give it a try.

*We are trying to specify SQLType used for parameter setter.* 
That is my point. The current implementation uses SQL Server types(`SSType.DECIMAL` ,`SSType.NUMERIC`) when throwing `R_valueOutOfRangeSQLType`. Using `dtv.getJdbcType()` will make sure we are using Java SQL types instead. 
Also notice that there is a slight difference too: Java SQL Type is `DECIMAL` whereas SQL Server type is `decimal`.